### PR TITLE
fix bug token params

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**
- fix bug token can not select by params if different chain, [detail](https://www.notion.so/kybernetwork/Referral-The-referral-link-works-incorrectly-74b3003ac5ad45189c923fd2c9b2193b)
=> solution: add conditions

### CODE REVIEW

File: src/components/Tutorial/TutorialSwap/index.tsx [hunk 1]
```diff
@@ -225,9 +225,9 @@ function Step3({ videoStyle = {} }: { videoStyle: CSSProperties }) {
           title="Tutorial kyberswap"
           allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
           allowFullScreen
-        ></iframe>
+        />
         {/** because we need tracking we user click video, iframe youtube not fire any event for us. */}
-        {!playedVideo && <TouchAbleVideo onClick={playVideo} />}
+        {!playedVideo && !isMobile && <TouchAbleVideo onClick={playVideo} />}
```
Comment (nguyenhoaidanh) [inline src/components/Tutorial/TutorialSwap/index.tsx:230]: Don't care about this. I just want to update UX tutorial swap on mobile.

File: src/pages/SwapV2/index.tsx [hunk 2]
```diff
@@ -629,17 +629,19 @@ export default function Swap({ history }: RouteComponentProps) {
     if (isSelectCurencyMannual) syncUrl(currencyIn, currencyOut) // when we select token manual
   }, [currencyIn, currencyOut, isSelectCurencyMannual, syncUrl])
 
+  const chainIdDebounce = usePrevious(chainId) || chainId
```
Comment (namgold) [inline src/pages/SwapV2/index.tsx:632]: name `Debounce` make me confuse. Name it `chainIdPrevious`

File: src/components/Tutorial/TutorialSwap/index.tsx (hunk 1 above)
Comment (hungdoansy) [reply src/components/Tutorial/TutorialSwap/index.tsx:230]: Why not track it on mobile? Is it a request from Product team?

File: src/pages/SwapV2/index.tsx (hunk 2 above)
Comment (nguyenhoaidanh) [reply src/pages/SwapV2/index.tsx]: ah, sorry. I forgot rename it. updated

File: src/components/Tutorial/TutorialSwap/index.tsx (hunk 1 above)
Comment (nguyenhoaidanh) [reply src/components/Tutorial/TutorialSwap/index.tsx:230]: yes, Follow spec.

File: src/pages/SwapV2/index.tsx [hunk 3]
```diff
@@ -629,17 +629,19 @@ export default function Swap({ history }: RouteComponentProps) {
     if (isSelectCurencyMannual) syncUrl(currencyIn, currencyOut) // when we select token manual
   }, [currencyIn, currencyOut, isSelectCurencyMannual, syncUrl])
 
+  const prevChainId = usePrevious(chainId) || chainId
+  const justChangeNetwork = prevChainId !== chainId
```
Comment (namgold) [inline src/pages/SwapV2/index.tsx:633]: This code made `justChangeNetwork` flip-flop between `true` and `false`, which `true` makes useEffect run, but `false` not and make useless re-render. I suggest make it a variable which count times of chainId changed `chainIdChangedCount` and useEffect gonna watch on that variable to run effect
```
useEffect(() => {
  ...
  // disable react exhaustive deps
}, [chainIdChangedCount])
```

Comment (hungdoansy) [reply src/pages/SwapV2/index.tsx:633]: I guess it only changes from `false` to `true` when switching networks 🤔

Comment (nguyenhoaidanh) [reply src/pages/SwapV2/index.tsx:633]: > I guess it only changes from `false` to `true` when switching networks 🤔

yes. correct

Comment (namgold) [reply src/pages/SwapV2/index.tsx:633]: then after that, one re-render happened which is not happened by change network, usePrevious return same chainId with current one, which make it change again from `true` to false`

File: src/pages/SwapV2/index.tsx [hunk 4]
```diff
@@ -629,17 +629,19 @@ export default function Swap({ history }: RouteComponentProps) {
...
   // swap?inputCurrency=xxx&outputCurrency=yyy. xxx yyy not exist in chain => remove params => select default pair
   useEffect(() => {
-    if (isPairNotfound) {
+    if (isPairNotfound && !currencyIn && !currencyOut && justChangeNetwork) {
```
Comment (namgold) [inline src/pages/SwapV2/index.tsx:636]: dont quite get it this `if` conditional. can you help explain

**Review by namgold — DISMISSED** [review]

**Review by hungdoansy — DISMISSED** [review]

**Review by hungdoansy — APPROVED** [review]

**Review by namgold — APPROVED** [review]

**Pull request merged**